adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java

```diff
@@ -282,11 +286,13 @@ public PathConfig matches(String targetUri) {
                             PolicyEnforcerConfig.EnforcementMode enforcementMode = PolicyEnforcerConfig.EnforcementMode.ENFORCING;
                             ResourceRepresentation targetResource = matchingResources.get(0);
                             List<PolicyEnforcerConfig.MethodConfig> methodConfig = null;
+                            boolean isStatic = false;
```

Comment:
@pedroigor I am not following the changes below this line. Why is this needed? 

Usecase where it doesn't make sense to me:
1. In `application.properties` file I have some static `pathConfig` that contains `*`  
2. Then for all requests that match the `pathConfig` I will set the `isStatic` to `true` even though the config is lazy loaded from server which means that probably should not be static. Am I missing something?  

Also, if I am not mistaken the reported issue in point 1. (from report #9855) should be fixed by change on line 200-202.

Reply:
A `static` path is a path that you explicitly associated with a resource. For those paths, the PEP is not going to invalidate the cache in order to sync changes in your resources (e.g.: changing the URI).

When you set `*` the server will first try to resolve the best match (exact or partial) otherwise use the path configuration you defined in the configuration.

The issue is about removing entries when they are marked as disabled (what should mark a path as `static` and avoid removing it from the paths) and not respecting the `static` marker when paths are configured as such. If you can elaborate more about the fix to line 200-202, I appreciate it. So I can see if I'm missing something.

Reply:
What I meant by `200-202` is this part: https://github.com/keycloak/keycloak/pull/13737/files#diff-0db76367dca5a7a87f67bf3a27f7e961e885b2361a1e768c44c1fa554a9b8036R200-R202

If I am not mistaken it solves the issue where disabled pathConfigs are removed from the cache itself.  What I don't understand is what is the purpose of the other changes in the `PolicyEnforcer` class.

This is a scenario I have on my mind when I see the other changes and I am not sure it is correct:
1. PolicyEnforcer config contains some path that is statically bound (P1)
2. There is a request that partially matches P1 therefore it is taken from the cache on this line
https://github.com/keycloak/keycloak/blob/e1fed2c95ca275aeee057a7fc0838ebc14324930/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java#L271
3. It is only a partial match, this means we will search Keycloak server for better matches and let's say we have found some resource (R1)
4. We create pathConfig based on R1:
https://github.com/keycloak/keycloak/blob/e1fed2c95ca275aeee057a7fc0838ebc14324930/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java#L306
5. And then set static flag based on P1
https://github.com/keycloak/keycloak/blob/e1fed2c95ca275aeee057a7fc0838ebc14324930/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java#L316
6. And then cache it and never removed because it is static
https://github.com/keycloak/keycloak/blob/e1fed2c95ca275aeee057a7fc0838ebc14324930/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java#L326

Is this correct? Or Am I missing something?

Reply:
As discussed, behavior expected.